sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/CachingFactory.java

```diff
@@ -31,7 +31,7 @@
  * significant for larger schemas) on each lookup. This wrapper caches the value returned by the
  * inner factory, so the schema comparison only need happen on the first lookup.
  */
-class CachingFactory<CreatedT> implements Factory<CreatedT> {
+public class CachingFactory<CreatedT> implements Factory<CreatedT> {
```

Comment:
Let's mark this `@Internal`, it's only public so we can use it from the arrow extension.